homeassistant/components/modbus/entity.py

```diff
@@ -181,8 +177,6 @@ def __init__(self, hass: HomeAssistant, hub: ModbusHub, config: dict) -> None:
             self._precision = config.get(CONF_PRECISION, 2)
         else:
             self._precision = config.get(CONF_PRECISION, 0)
-            if self._precision > 0 or self._scale != int(self._scale):
-                self._value_is_int = False
```

Comment:
Removing precision here, might be correct, but it seems it is not added to climate ??